src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -156475,5 +156451,243 @@ class A : Attribute
+                //         (default(A<string?>)).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "default(A<string?>)").WithLocation(12, 10),
+                // (13,10): warning CS8602: Dereference of a possibly null reference.
+                //         (default(A<string?>?)).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "default(A<string?>?)").WithLocation(13, 10));
+        }
+
+        [Fact, WorkItem(63425, "https://github.com/dotnet/roslyn/issues/63425")]
+        public void DefaultOfT4()
+        {
+            var comp = CreateCompilation(new[] { """
+#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+class C<T>
+{
+    void M1()
+    {
+        if (default(T) is not null)
+        {
+            M2(default);
+            M3(default);
+        }
+    }
+
+    void M2()
+    {
+        M2(default); // 1
+        M3(default);
+    }
+
+    void M3()
+    {
+        M3(default); // 2
+        M2(default);
+    }
+
+    void M2(T t) { }
+    void M3([DisallowNull] T t) { }
+}
+""", DisallowNullAttributeDefinition });
+            comp.VerifyDiagnostics(
+                // (18,12): warning CS8604: Possible null reference argument for parameter 't' in 'void C<T>.M2(T t)'.
+                //         M2(default); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "default").WithArguments("t", "void C<T>.M2(T t)").WithLocation(18, 12),
+                // (24,12): warning CS8604: Possible null reference argument for parameter 't' in 'void C<T>.M3(T t)'.
+                //         M3(default); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "default").WithArguments("t", "void C<T>.M3(T t)").WithLocation(24, 12));
+        }
+
+        [Fact, WorkItem(63425, "https://github.com/dotnet/roslyn/issues/63425")]
+        public void DefaultOfT5()
+        {
+            var comp = CreateCompilation(new[] { """
+#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+class C<T>
+{
+    void AssertNotNull([NotNull] T t) { }
+
+    void M1()
+    {
+        AssertNotNull(default);
+        default(T).ToString(); // ok
+    }
+}
+""", NotNullAttributeDefinition });
+            comp.VerifyDiagnostics(
+                // (7,41): warning CS8777: Parameter 't' must have a non-null value when exiting.
+                //     void AssertNotNull([NotNull] T t) { }
+                Diagnostic(ErrorCode.WRN_ParameterDisallowsNull, "}").WithArguments("t").WithLocation(7, 41),
+                // (11,23): warning CS8604: Possible null reference argument for parameter 't' in 'void C<T>.AssertNotNull(T t)'.
+                //         AssertNotNull(default);
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "default").WithArguments("t", "void C<T>.AssertNotNull(T t)").WithLocation(11, 23),
+                // NOTE: This warning is incorrect. See https://github.com/dotnet/roslyn/issues/63488
```

Comment:
I think we can fix this in this PR. All it should take is, if we are asked for the slot for a default literal, we should be willing to create one if it doesn't already exist.

Reply:
It seems we create the slot already when we learn from `NotNull` and set the state correctly:

![image](https://user-images.githubusercontent.com/31348972/201200438-11b63666-167b-490a-a206-33f34b0764d7.png)

I wasn't able to dig further.